tests/test_routes.py

```diff
@@ -0,0 +1,91 @@
+from flask import session
+from urlshortener.extensions import mongo
+
+
+def test_home(client):
+    response = client.get("/")
+
+    assert response.status_code == 200
+    assert b"<title>URL Shortener</title>" in response.data
+
+
+def test_add(client):
+    client.post("/add", data={"long_url": "https://www.url.test"})
+
+    assert mongo.db.links.estimated_document_count() == 1
+    assert len(mongo.db.links.find_one()["short_url"]) == 3
```

Comment:
I don't think we strictly validate types anymore? That this would be fulfilled both by [1,2,3] and "123". I would recommend asserting so that we know what type this is. Alternatively, we could assert with a regex.